[ews-build.webkit.org] Switch macOS JSC arm64 stress test queue to O3 debug builds

ryanhaddad opened the pull request:
#### f19f9cb181c1bc9961f995da9b342c216300b8d3
<pre>
[ews-build.webkit.org] Switch macOS JSC arm64 stress test queue to O3 debug builds
<a href="https://rdar.apple.com/159975164">rdar://159975164</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=303943">https://bugs.webkit.org/show_bug.cgi?id=303943</a>

Reviewed by Keith Miller.

This adds an assert-enabled JSC stress test configuration to pre-commit testing so that regressions
can be caught before landing. This configuration has been running successfully in post-commit
for the past few weeks.

Switching to O3 debug builds adds ~20 minutes to test runtime on our existing arm64 EWS hardware
(51m vs. 31m on release builds). The x86 configuration will not change at this time.

* Tools/CISupport/Shared/steps.py:
(SetO3OptimizationLevel): Move this step to share it between ews-build and build.webkit.og
* Tools/CISupport/build-webkit-org/steps.py:
(SetO3OptimizationLevel): Deleted.
* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS): Update queue name to reflect the configuration change.
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble): Ditto.
* Tools/CISupport/ews-build/config.json: Ditto.
* Tools/CISupport/ews-build/factories.py:
(JSCBuildO3AndTestsFactory): Add a new factory that includes the step to set the O3 optimization level.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Update unit tests.
* Tools/CISupport/ews-build/loadConfig.py: Import the SetO3OptimizationLevel step.
* Tools/CISupport/ews-build/steps.py: Ditto.
(RunJavaScriptCoreTests.evaluateCommand): Run the SetO3OptimizationLevel step before re-building
without the PR.

Canonical link: <a href="https://commits.webkit.org/305715@main">https://commits.webkit.org/305715@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f82164759fbdd91242fa0ae08a210d23e9b4762c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/139104 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/11477 "Built successfully") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/147231 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/92171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/140976 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/12184 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/11628 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/106483 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/92171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/142051 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/9201 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/87351 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/8754 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/6530 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/7523 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/118203 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/150010 "Built successfully") | | 
| [⏳ 🛠 🧪 jsc-debug-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-O3-Debug-arm64-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/11162 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/114873 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/138529 "Passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/11175 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/9441 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/115185 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/66064 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21461 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/11204 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/168/builds/601 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/10940 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/11143 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/10992 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 reviewed (approved):
r=me probably still needs a review from someone more familiar with the CI code than me though.

ryanhaddad committed:
[ews-build.webkit.org] Switch macOS JSC arm64 stress test queue to O3 debug builds
rdar://159975164
https://bugs.webkit.org/show_bug.cgi?id=303943

Reviewed by Keith Miller.

This adds an assert-enabled JSC stress test configuration to pre-commit testing so that regressions
can be caught before landing. This configuration has been running successfully in post-commit
for the past few weeks.

Switching to O3 debug builds adds ~20 minutes to test runtime on our existing arm64 EWS hardware
(51m vs. 31m on release builds). The x86 configuration will not change at this time.

* Tools/CISupport/Shared/steps.py:
(SetO3OptimizationLevel): Move this step to share it between ews-build and build.webkit.og
* Tools/CISupport/build-webkit-org/steps.py:
(SetO3OptimizationLevel): Deleted.
* Tools/CISupport/ews-app/ews/common/github.py:
(GitHubEWS): Update queue name to reflect the configuration change.
* Tools/CISupport/ews-app/ews/views/statusbubble.py:
(StatusBubble): Ditto.
* Tools/CISupport/ews-build/config.json: Ditto.
* Tools/CISupport/ews-build/factories.py:
(JSCBuildO3AndTestsFactory): Add a new factory that includes the step to set the O3 optimization level.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Update unit tests.
* Tools/CISupport/ews-build/loadConfig.py: Import the SetO3OptimizationLevel step.
* Tools/CISupport/ews-build/steps.py: Ditto.
(RunJavaScriptCoreTests.evaluateCommand): Run the SetO3OptimizationLevel step before re-building
without the PR.

Canonical link: https://commits.webkit.org/305715@main
